packages/engine/Source/Scene/Cesium3DTileset.js

```diff
@@ -2661,43 +2793,26 @@ function handleTileFailure(error, tileset, tile) {
   }
 }
 
-/**
- * @private
- * @param {Cesium3DTileset} tileset
- */
-function filterProcessingQueue(tileset) {
-  const tiles = tileset._processingQueue;
-
-  let removeCount = 0;
-  for (let i = 0; i < tiles.length; ++i) {
-    const tile = tiles[i];
-    if (
-      tile.isDestroyed() ||
-      tile._contentState !== Cesium3DTileContentState.PROCESSING
-    ) {
-      ++removeCount;
-      continue;
-    }
-    if (removeCount > 0) {
-      tiles[i - removeCount] = tile;
-    }
-  }
-  tiles.length -= removeCount;
-}
-
 /**
  * Process tiles in the PROCESSING state so they will eventually move to the READY state.
  * @private
  * @param {Cesium3DTileset} tileset
  * @param {Cesium3DTile} tile
  */
 function processTiles(tileset, frameState) {
-  filterProcessingQueue(tileset);
-  const tiles = tileset._processingQueue;
-  const statistics = tileset._statistics;
-  let tile;
+  const tiles = tileset._processingQueue.filter(
+    (tile) =>
+      !tile.isDestroyed() &&
+      tile._contentState === Cesium3DTileContentState.PROCESSING
+  );
```

Comment:
While more concise, this will allocate an array every frame (which is what `filterProcessingQueue` avoids)

Reply:
I restored `filterProcessingQueue` for now.

We are also potentially re-allocating by using `tileset._processingQueue.push()` in `requestContent`. If this is really an issue, perhaps a separate PR could convert `._processingQueue` to a `ManagedArray`, and move the custom filter function to a method of `ManagedArray`.